common/motion_utils/include/motion_utils/trajectory/trajectory.hpp

```diff
@@ -1245,6 +1245,46 @@ inline boost::optional<size_t> insertStopPoint(
   return stop_idx;
 }
 
+/**
+ * @brief Insert deceleration point from the source pose
+ * @param src_point source point
+ * @param distance_to_decel_point  distance to deceleration point from the src point
+ * @param velocity velocity of stop point
+ * @param points_with_twist output points of trajectory, path, ... (with velocity)
+ */
+template <class T>
+boost::optional<size_t> insertDecelPoint(
+  const geometry_msgs::msg::Point & src_point, const double distance_to_decel_point,
+  const double velocity, T & points_with_twist)
+{
+  const auto decel_point =
+    calcLongitudinalOffsetPoint(points_with_twist, src_point, distance_to_decel_point);
+
+  if (!decel_point) {
+    return {};
+  }
+
+  const auto seg_idx = findNearestSegmentIndex(points_with_twist, decel_point.get());
+  const auto insert_idx = insertTargetPoint(seg_idx, decel_point.get(), points_with_twist);
+
+  if (!insert_idx) {
+    return {};
+  }
+
+  const auto insertVelocity = [&insert_idx](T & points_with_twist, const float v) {
+    for (size_t i = insert_idx.get(); i < points_with_twist.size(); ++i) {
+      const auto & original_velocity =
+        tier4_autoware_utils::getLongitudinalVelocity(points_with_twist.at(i));
+      tier4_autoware_utils::setLongitudinalVelocity(
+        std::min(static_cast<float>(original_velocity), v), points_with_twist.at(i));
+    }
+  };
```

Comment:
Do you need to make this a function? I think this is not necessary

Reply:
it's just copy from @satoshi-ota's function. but it seems no reason.
fixed in https://github.com/autowarefoundation/autoware.universe/pull/3226/commits/f0214b725b1a5bb2cfb05638f398efc9ddb1cc2c

Reply:
FYI: https://github.com/autowarefoundation/autoware.universe/blob/feat/insertDecelPoint/planning/behavior_path_planner/src/util/avoidance/util.cpp#L1000-L1007